engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -25,66 +25,79 @@
 import java.util.Map;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.terasology.context.Context;
 import org.terasology.engine.SimpleUri;
+import org.terasology.rendering.dag.gsoc.NewAbstractNode;
+import org.terasology.rendering.dag.gsoc.NewNode;
 
 /**
  * TODO: Add javadocs
  */
 public class RenderGraph {
     private static final Logger logger = LoggerFactory.getLogger(RenderGraph.class);
 
-    private Map<SimpleUri, Node> nodeMap;
-    private MutableGraph<Node> graph;
+    private Map<SimpleUri, NewNode> nodeMap;
+    private MutableGraph<NewNode> graph;
+    private Context context;
 
-    public RenderGraph() {
+    public RenderGraph(Context context) {
         nodeMap = Maps.newHashMap();
         graph = GraphBuilder.directed().build();
+        this.context = context;
```

Comment:
Can you please tell me where this variable is used?

Reply:
I still haven't decided where to store context to use in nodes...we'll see what I wind up with after figuring out how to merge some classes. At this point though it is probably redundant (Node.setDependencies() - node has the context)